components/style/rule_tree/mod.rs

```diff
@@ -728,6 +759,13 @@ impl StrongRuleNode {
             self.gc();
         }
     }
+
+    /// Returns true if there is either animation or transition level rule.
+    pub fn has_animation_or_transition_rules(&self) -> bool {
+        self.self_and_ancestors().take_while(|node| node.cascade_level() >= CascadeLevel::Animations)
+                                 .any(|node| node.cascade_level() == CascadeLevel::Animations ||
```

Comment:
consider indenting the `take_while` and `any` with the `self_and_ancestors`. Also, I believe you can use the `matches!` macro.

```rust
self.self_and_ancestors()
    .take_while(|node| node.cascade_level() >= CascadeLevel::Animations)
    .any(|node| matches!(node.cascade_level(), CascadeLevel::Animations | CascadeLevel::Transitions))
```